evm/vm/forks/sharding/contracts/validator_manager.v.py

```diff
@@ -363,21 +266,28 @@ def add_header(header: bytes <= 4096) -> bool:
         self.shard_head[shard_id] = entire_header_hash
         is_new_head = True
 
-    # Emit log
-    self.emit_collation_added(shard_id, header, is_new_head, _score)
+    # Emit a log which is equivalent to
+    # CollationAdded: __log__({shard_id: indexed(num), collation_header: bytes <= 4096, is_new_head: bool, score: num})
+    # TODO: should be replaced by `log.CollationAdded`
+    if is_new_head:
+        new_head_in_num = 1
+    else:
+        new_head_in_num = 0
+    raw_log(
+        [
+            sha3("CollationAdded(int128,bytes4096,bool,int128)"),
```

Comment:
Yes, actually it matches log signature `CollationAdded(indexed int128 shard, bytes4096 collationHeader, bool isNewHead, int128 score)`.
The reason why I use `int128`(i.e. `num` in vyper) is it is more friendly for calculation. And the reason why I use `raw_log` instead of `log.CollationAdded()` is previously vyper doesn't support bytes type whose length > 32. Actually there is a TODO comment in line 67

Reply:
Can the `collationHeader` argument from `CollationAdded` be unpacked?

So instead of using `CollationAdded(indexed int128 shard, bytes4096 collationHeader, bool isNewHead, int128 score)`
Use `CollationAdded(indexed int128 shard, int128 expectedPeriodNumber, bytes32 periodStartPrevHash, bytes32 _parentCollationHash, bytes32 txListRoot, address collationCoinbase, bytes32 postStateRoot, bytes32 receiptRoot, int128 collationNumber, bool isNewHead, int128 score)`

The benefits would be: 1 int less for the log: `shard`; easier to use with the ABI.

Reply:
I think it's a good point. I will see if there are specific reason for using `CollationAdded(indexed int128 shard, bytes4096 collationHeader, bool isNewHead, int128 score)`.
Thank you for the suggestion.